src/main/java/com/google/cloud/datafusion/plugin/successfactors/source/metadata/SuccessFactorsSchemaGenerator.java

```diff
@@ -0,0 +1,770 @@
+      if (type instanceof EdmProperty) {
+        EdmProperty edmProperty = (EdmProperty) type;
+        successFactorsColumnDetailList.add(buildSuccessFactorsColumnMetadata(namespace, edmProperty));
+
+      }
+
+      // check for navigation property
+      if (type instanceof EdmNavigationPropertyImplProv) {
+
+        EdmNavigationPropertyImplProv navProperty = (EdmNavigationPropertyImplProv) type;
+        EdmEntityType navEntityType = successFactorsServiceHelper.extractEntitySetFromNavigationProperty(navProperty);
+        if (navEntityType != null) {
+
+          List<SuccessFactorsColumnMetadata> navChild = buildSuccessFactorsColumns(navEntityType,
+                                                                             navEntityType.getPropertyNames());
+
+          SuccessFactorsColumnMetadata navigationColumn = SuccessFactorsColumnMetadata.builder()
+            .name(prop)
+            .type(navProperty.getType().getName())
+            .multiplicityOrdinal(navProperty.getMultiplicity().ordinal())
+            .childList(navChild)
+            .build();
+
+          successFactorsColumnDetailList.add(navigationColumn);
+        }
+      }
+    }
+
+    return successFactorsColumnDetailList;
+  }
+
+  /**
+   * Builds the {@code SuccessFactorsColumnMetadata} from the given {@code EdmProperty}.
+   * Also builds the COMPLEX property.
+   *
+   * @param namespace   SAP SuccessFactors entity namespace. Used to build the COMPLEX properties.
+   * @param edmProperty {@code EdmProperty} of the SuccessFactors service entity.
+   * @return {@code SuccessFactorsColumnMetadata}
+   * @throws EdmException any apache olingo processing exception
+   */
+  private SuccessFactorsColumnMetadata buildSuccessFactorsColumnMetadata(String namespace, EdmProperty edmProperty)
+    throws EdmException {
+
+    SuccessFactorsColumnMetadata.Builder successFactorsColumnDetailBuilder = SuccessFactorsColumnMetadata.builder()
+      .name(edmProperty.getName())
+      .kindName(edmProperty.getType().getKind().name())
+      .type(edmProperty.getType().getName())
+      .multiplicityOrdinal(edmProperty.getMultiplicity().ordinal());
+
+    if (edmProperty.getFacets() != null) {
+      EdmFacets facets = edmProperty.getFacets();
+      successFactorsColumnDetailBuilder
+        .collation(facets.getCollation())
+        .defaultValue(facets.getDefaultValue())
+        .maxLength(facets.getMaxLength())
+        .precision(facets.getPrecision())
+        .scale(facets.getScale())
+        .isFixedLength(facets.isFixedLength())
+        .isUnicode(facets.isUnicode())
+        .isNullable(facets.isNullable());
+    }
+
+    //setting SAP specific details.
+    List<EdmAnnotationAttribute> edmAnnotationAttribute = edmProperty.getAnnotations().getAnnotationAttributes();
+    if (edmAnnotationAttribute != null) {
+      edmAnnotationAttribute.forEach(sapAttribute -> {
+        switch (sapAttribute.getName()) {
+          case "display-format":
```

Comment:
The magic strings for the switch cases looks error-prone. Do we have any Enums that we can use instead of defining these strings ourselves? That way if SAP's implementation changes we won't have to keep changing accordingly

Reply:
Added `SapAttributes` constants file to store the magic strings. Tried enum but case statements have to be compile-time evaluable, so something like `SapAttributes.DISPLAY_FORMAT.toString()` didn't satisfy. A string literal, on the other hand, did. https://github.com/data-integrations/successfactors-plugins/commit/0ee84795bfe46404b5b855b83136b895898b170d